src/customer/strategies/masterpass-customer-strategy.ts

```diff
@@ -0,0 +1,133 @@
+                if (!cart) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCart);
+                }
+
+                const { container, buttonId } = masterpassOptions;
+                this._buttonId = (buttonId ? buttonId : 'mpbutton');
+
+                const payload = {
+                    checkoutId: this._paymentMethod.initializationData.checkoutId,
+                    allowedCardTypes: this._paymentMethod.initializationData.allowedCardTypes,
+                    amount: cart.cartAmount.toFixed(2).toString(),
+                    currency: cart.currency.code,
+                    cartId: cart.id,
+                };
+
+                return this._masterpassScriptLoader.load(this._paymentMethod.config.testMode)
+                    .then(Masterpass => {
+                        const signInButton = this._createSignInButton(container);
+
+                        signInButton.onclick = () => {
+                            Masterpass.checkout(payload);
+                        };
+                    });
+            })
+            .then(() => super.initialize(options));
+    }
+
+    deinitialize(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this._isInitialized) {
+            return super.deinitialize(options);
+        }
+
+        this._paymentMethod = undefined;
+        if (this._buttonId !== undefined) {
+            const signInButton = document.getElementById(this._buttonId);
+            if (signInButton !== null && signInButton.parentNode !== null) {
+                signInButton.onclick = null;
+                signInButton.parentNode.removeChild(signInButton);
+            }
+        }
+
+        return super.deinitialize(options);
+    }
+
+    signIn(credentials: CustomerCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        throw new NotImplementedError(
+            'In order to sign in via Masterpass, the shopper must click on "Masterpass" button.'
+        );
+    }
+
+    signOut(options?: any): Promise<InternalCheckoutSelectors> {
+        const state = this._store.getState();
+        const payment = state.payment.getPaymentId();
+
+        if (!payment) {
+            return Promise.resolve(this._store.getState());
+        }
+
+        return this._store.dispatch(
+            this._remoteCheckoutActionCreator.signOut(payment.providerId, options)
+        );
+    }
+
+    private _createSignInButton(containerId: string): HTMLElement {
+        const container = document.querySelector(`#${containerId}`);
+
+        if (!container) {
+            throw new Error('Need a container to place the button');
+        }
+
+        return this._insertMasterpassButton(container);
+    }
+
+    private _insertMasterpassButton(container: Element): HTMLElement {
+        const buttonTemplate = `
+            <img
+            id="${this._buttonId}"
+            style="cursor: pointer;"
```

Comment:
Instead of setting `style="cursor: pointer;"` to an image tag, you should consider wrapping the image with an anchor or a button tag so that it is keyboard accessible.